### ISSUE TITLE
unbound local variable detection

### ISSUE THREAD

**Issue by thisch** [PyCQA/pylint#1838]
### Steps to reproduce
1. test the following module with pylint 
```py
def foo(expr):
    if expr:
        a, b = 3, 4
    else:
        b = 2
    return a, b
```

### Current behavior

```
> pylint demo.py 
No config file found, using default configuration
************* Module demo
C:  1, 0: Missing module docstring (missing-docstring)
C:  1, 0: Black listed name "foo" (blacklisted-name)
C:  1, 0: Missing function docstring (missing-docstring)
C:  3, 8: Invalid variable name "a" (invalid-name)
C:  3,11: Invalid variable name "b" (invalid-name)
C:  6, 8: Invalid variable name "b" (invalid-name)
```

### Expected behavior

I expect sth like the following in the output of pylint

```
E:  7, 12: Local variable "a" might be reference before assignment (reference-before-assignment)
```

### pylint --version output

```
No config file found, using default configuration
pylint 1.5.2, 
astroid 1.4.4
Python 2.7.12 (default, Nov 20 2017, 18:23:56) 
[GCC 5.4.0 20160609]
```

**Comment by brycepg** [comment]
I think this is a duplicate of #1807 , which requires pylint to understand control flow

**Comment by PCManticore** [comment]
Yeah, @brycepg is right. I'm going to close this a duplicate. Even if we could fix this without control flow, it would add too many special cases through out the code base.
